src/ui/public/dialogues/confirm_dialogue.js

```diff
@@ -0,0 +1,63 @@
+import uiModules from 'ui/modules';
+import confirmDialogueTemplate from './confirm_dialogue.html';
+
+const module = uiModules.get('kibana');
+
+module.factory('showConfirmDialogue', function (ModalDialogue, $q, $compile, $rootScope) {
+  /**
+   * Shows a modal confirmation dialog with the given message.
+   *
+   * @param {String} message
+   * @param {String} yesButtonText - Defaults to Okay, supply a custom string to override.
+   * @param {String} noButtonText - Defaults to Cancel, supply a custom string to override.
+   * @return {Promise<boolean>} Returns a promise that will be resolved to true if the user
+   * clicks the yes/okay button and rejected if the user clicks the no/cancel button.
+   */
+  return function showConfirmDialogue(message, yesButtonText = 'Okay', noButtonText = 'Cancel') {
```

Comment:
Let's remove the default for the `yesButtonText`. Confirmation actions should always explicitly state what will happen to help clarify the consequences for the user.

It shouldn't be hard to update the rest of Kibana to reflect this change since it looks like it's only used in a couple other places.